core/coreapi/coreapi.go

```diff
@@ -92,3 +95,10 @@ func (api *CoreAPI) getSession(ctx context.Context) *CoreAPI {
 	ng := dag.NewReadOnlyDagService(dag.NewSession(ctx, api.dag))
 	return &CoreAPI{api.node, ng}
 }
+
+// Offline returns new api with offline DAG
+func (api *CoreAPI) Offline() *CoreAPI {
+	bs := api.node.Blocks.Blockstore()
+	ng := dag.NewDAGService(bserv.New(bs, offline.Exchange(bs)))
```

Comment:
I'm really worried we're going to bypass this dag accidentally by calling something on the node that has the wrong dag. Unfortunately, I'm not sure if there's a better solution (that works with sessions, anyways, offline can be done with a context hint).

Reply:
But I'm probably just paranoid.

Reply:
Yeah, I'll probably go with context hints (#5654), but that's a larger change